neuralcompression/models/_unet_discriminator.py

```diff
@@ -0,0 +1,337 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from functools import partial
+from typing import Callable, List, Optional, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn.utils.parametrizations import spectral_norm
+
+from ._discriminator import Discriminator
+
+
+def _verify_norm_type(norm_type: str):
+    if norm_type not in ("batch", "spectral", "instance", "identity"):
+        raise ValueError(f"Unrecognized norm_type {norm_type}")
+
+
+def apply_conv_norm(module: nn.Module, norm_type: str = "identity") -> nn.Module:
+    _verify_norm_type(norm_type)
+    if norm_type == "spectral":
+        return spectral_norm(module)
+    else:
+        return module
+
+
+def build_norm_layer(chans: int, norm_type: str) -> nn.Module:
```

Comment:
In Omnivision, we avoid this kind of functions by requiring targets in configuration files (but it is waaay more verbose)

Reply:
Oh I do think that would be better, but might put too much hydra into here.